### PR TITLE
feat: support `node:` prefix for Node.js core modules in runtime code

### PR DESCRIPTION
**Pull request by alexander-akait**
**What kind of change does this PR introduce?**

fixes https://github.com/webpack/webpack/issues/18277

**Did you add tests for your changes?**

Yes

**Does this PR introduce a breaking change?**

No, because we generate it only when your env supported it

**What needs to be documented once your changes are merged?**

Yes

Shorty:

We generate `node:` prefix for `import ... from "module"` only when your env supported it and it is only abour webpack runtime code. We need document that you can disable it using the `output.environment.nodePrefixForCoreModules` option

### CODE REVIEW

**Comment by alexander-akait** [comment]
@vankop I want to drop azure CI, it is slow and often failed, what do you think?

**Comment by vankop** [comment]
you are right, but ci speed in general slow.. maybe keep it for stability, we can force merge anyway.

File: lib/config/target.js [hunk 1]
```diff
@@ -168,8 +169,8 @@ You can also more options via the 'target' option: 'browserslist' / 'browserslis
 	[
 		"[async-]node[X[.Y]]",
 		"Node.js in version X.Y. The 'async-' prefix will load chunks asynchronously via 'fs' and 'vm' instead of 'require()'. Examples: node14.5, async-node10.",
-		/^(async-)?node(\d+(?:\.(\d+))?)?$/,
-		(asyncFlag, major, minor) => {
+		/^(async-)?node((\d+)(?:\.(\d+))?)?$/,
+		(asyncFlag, _, major, minor) => {
```
Comment (alexander-akait) [inline lib/config/target.js:173]: Bugfix, before when you have `14.18` - major was `14.18` and minor was `18`, so if you have `node14.8` in your target, you always have true, same here for other places

Comment (vankop) [reply lib/config/target.js:173]: how this work before 🙈

Comment (alexander-akait) [reply lib/config/target.js:173]: Looks like nobody use it before :smile:

**Pull request merged**